src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs

```diff
@@ -3308,13 +3305,13 @@ private void CheckAllowContent()
             map[(int)BinXmlToken.SQL_UDT] = TypeOfByteArray;
             map[(int)BinXmlToken.XSD_BINHEX] = TypeOfByteArray;
             map[(int)BinXmlToken.XSD_BASE64] = TypeOfByteArray;
-            map[(int)BinXmlToken.SQL_CHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_VARCHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_TEXT] = TypeOfString;
-            map[(int)BinXmlToken.SQL_NCHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_NVARCHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_NTEXT] = TypeOfString;
-            map[(int)BinXmlToken.SQL_UUID] = TypeOfString;
+            map[(int)BinXmlToken.SQL_CHAR] = typeof(string);
```

Comment:
You can keep `TypeOfString` here, just change it to a local - similar to how the other similar TypeOf... in this method are done.